crypto/bls/mod.go

```diff
@@ -14,119 +16,144 @@ import (
 
 //go:generate protoc -I ./ --go_out=./ ./messages.proto
 
-var suite = pairing.NewSuiteBn256()
+var (
+	suite = pairing.NewSuiteBn256()
+
+	protoenc encoding.ProtoMarshaler = encoding.NewProtoEncoder()
+)
 
+// publicKey is the implementation of the crypto.PublicKey interface for a BLS
```

Comment:
For a later refactoring, I suggest we systematically use `<element> implements <abstraction>(). It ...` if its an interface function and `<element> implements <abstraction>{}. It ...` if it's a struct.